File: 392.is-subsequence.cpp
class Solution {
public:
    bool isSubsequence(string s, string t) {
        if(s=="") return true;
        else if(t=="") return false;
        for(int i=0;i<t.length();i++)
        {
            if(t[i]==s[0])
            {
                return isSubsequence(s.substr(1),t.substr(i+1));
            }
        }
        return false;
    }
};

File: 392.is-subsequence.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        if s is "":
            return True
        l, r = 0 , len(s)
        i = 0
        while i < (len(t)):
            while t[i] == s[l]:
                l+=1
                i+=1
                if l == r:
                    return True
                if i == len(t):
                    return False
            i += 1
        
        return False

        